xcell/cls/cl.py

```diff
@@ -261,160 +261,22 @@ def __init__(self, data, tr1, tr2):
 
         self.outdir = self.get_outdir('fiducial')
         os.makedirs(self.outdir, exist_ok=True)
-        self.cosmo = self.get_cosmo_ccl()
-        self.hm_par = self._get_halomodel_params()
+        self.th = Theory(self.data.data)
```

Comment:
Why don't you make `ClFid` a child also of Theory? (i.e. `class ClFid(ClBase, Theory)`)?

Reply:
Fancy! Let me see if I can make that work...

Reply:
Looking into it I'm not sure this is a good idea. For this to work, `ClBase` and `Theory` would need to have the same `__init__()` syntax, which they don't. I'd rather not force the code to do this just for the sake of multiple inheritance fanciness.